controllers/networkplugins_controller.go

```diff
@@ -17,35 +17,584 @@ limitations under the License.
+	if err != nil {
+		fmt.Printf("ERROR PARSEFILE: %v\n", err)
+		return err
+	}
+
+	if err := renderTemplateToFile(config, t, outputDir+"sriov-cni.yaml"); err != nil {
+		return err
+	}
+
+	// Apply the SRIOV Device Plugin
+	t, err = template.ParseFiles(filepath.Join(TemplateDir, "sriov", "sriov-deviceplugin.yaml"))
+	if err != nil {
+		fmt.Printf("ERROR PARSEFILE: %v\n", err)
+		return err
+	}
+
+	if err := renderTemplateToFile(config, t, filepath.Join(outputDir, "sriov-deviceplugin.yaml")); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (sriovConfig *SriovT) ApplyTemplate(outputDir string) error {
+	fmt.Println("Applying SRIOV")
+	return nil
+}
+
+func (r *NetworkPluginsReconciler) createPlugin(plugin ApplyPlugin) error {
+	outputDir := CreateDir
+
+	if err := plugin.WriteConfigToTemplate(outputDir); err != nil {
+		return err
+	}
+
+	if err := plugin.ApplyTemplate(outputDir); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (r *NetworkPluginsReconciler) deletePlugin(plugin ApplyPlugin) error {
+	outputDir := DeleteDir
+
+	if err := plugin.WriteConfigToTemplate(outputDir); err != nil {
+		return err
+	}
+
+	if err := plugin.ApplyTemplate(outputDir); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (r *NetworkPluginsReconciler) parseNewPlugins(fileList *[]string) error {
+	var spec *plumberv1.NetworkPluginsSpec = r.currentSpec
+
+	if err := os.MkdirAll(CreateDir, os.ModePerm); err != nil {
+		return err
+	}
+
+	if plugins := spec.CniPlugins; plugins != nil {
+		if plugins.Multus != nil {
+			multusConfig := (*MultusT)(plugins.Multus)
+			err := r.createPlugin(multusConfig)
+			if err != nil {
+				return err
+			}
+			*fileList = append(*fileList, "multus.yaml")
+		}
+
+		if plugins.Sriov != nil {
+			sriovConfig := (*SriovT)(plugins.Sriov)
+			err := r.createPlugin(sriovConfig)
+			if err != nil {
+				return err
+			}
+			*fileList = append(*fileList, "sriov-cni.yaml")
+			*fileList = append(*fileList, "sriov-deviceplugin.yaml")
+		}
+
+		if plugins.Whereabouts != nil {
+			whConfig := (*WhereaboutsT)(plugins.Whereabouts)
+			err := r.createPlugin(whConfig)
+			if err != nil {
+				return err
+			}
+			*fileList = append(*fileList, "whereabouts.yaml")
+		}
+
+		if plugins.HostPlumber != nil {
+			hostPlumberConfig := (*HostPlumberT)(plugins.HostPlumber)
+			err := r.createPlugin(hostPlumberConfig)
+			if err != nil {
+				return err
+			}
+			*fileList = append(*fileList, "hostplumber.yaml")
+		}
+
+		if plugins.NodeFeatureDiscovery != nil {
+			nfdConfig := (*NodeFeatureDiscoveryT)(plugins.NodeFeatureDiscovery)
+			err := r.createPlugin(nfdConfig)
+			if err != nil {
+				return err
+			}
+			*fileList = append(*fileList, "nfd.yaml")
+		}
+	}
+	return nil
+}
+
+func (r *NetworkPluginsReconciler) parseOldPlugins(fileList *[]string) error {
+	// First find out which plugins are missing from new spec vs old spec
+	if r.prevSpec == nil || r.prevSpec.CniPlugins == nil {
```

Comment:
If you want to be more kubernetes-native with this approach. Instead of storing the prevSpec in a configmap or somewhere and then retrieving it to compare to the new spec, use the status instead. 

The status should contain the state as it is right now in the cluster. The spec should contain the desired/new state. So the status would contain basically the `prevSpec`, which you compare with the `currentSpec` in the spec. And based on that you decide which plugins to install/delete.